src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs

```diff
@@ -271,6 +274,11 @@ public sealed class FailurePoint
             /// </summary>
             public long Position;
 
+            /// <summary>
+            /// The name of the property.
+            /// </summary>
+            public string? PropertyName;
```

Comment:
Could this be just a string, default empty string, to avoid nulls.

Reply:
It could, but it would mean that all places that create a FailurePoint get an (implicit) unnecessary extra initialization.
I find a check for `null` clearer than one for `string.IsNullOrEmpty`.
Also the compiler tells me when I don't to the `null` check, not when I miss the empty check.

Non-nullable is good for Collectons, so I updated the nullable failurePoint you mentioned where